File: SFA-Client/src/app/_models/salesRep.ts
export class SalesRep {
  id: number;
  userId: number;
  userName: string;
  salesRepCode: string;
  assignedStoreId: number;
  assignedStoreName: string;
  assignedAreaId: number;
  assignedAreaName: string;

  constructor(salesRep) {
    this.id = salesRep.id;
    this.userId = salesRep.userId;
    this.userName = salesRep.userName;
    this.salesRepCode = salesRep.salesRepCode;
    this.assignedStoreId = salesRep.assignedStoreId;
    this.assignedStoreName = salesRep.assignedStoreName;
    this.assignedAreaId = salesRep.assignedAreaId;
    this.assignedAreaName = salesRep.assignedAreaName;
  }
}

File: SFA-API/Domain/SalesRep.cs
using System.Collections.Generic;

namespace Domain
{
    public class SalesRep
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string SalesRepCode { get; set; }
        public int AssignedAreaId { get; set; }
        public int AssignedStoreId { get; set; }
        public virtual UserDetail UserDetail { get; set; }
    }
}